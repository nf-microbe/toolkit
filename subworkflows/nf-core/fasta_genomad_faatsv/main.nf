// Import modules
include { GENOMAD_DOWNLOAD  } from '../../../modules/nf-core/genomad/download'
include { GENOMAD_ENDTOEND  } from '../../../modules/nf-core/genomad/endtoend'

workflow FASTA_GENOMAD_FAATSV {
    take:
    fasta_gz            // [ [ meta ], fasta.gz ]
    genomad_db          // [ genomad_db ]

    main:
    ch_versions             = Channel.empty()
    ch_workdirs_to_clean    = Channel.empty()

    // if genomad_db exists, skip GENOMAD_DOWNLOAD
    if (genomad_db){
        ch_genomad_db = genomad_db
    } else {
        //
        // MODULE: download geNomad database
        //
        GENOMAD_DOWNLOAD(
        )
        ch_genomad_db   = GENOMAD_DOWNLOAD.out.genomad_db
        ch_versions     = ch_versions.mix(GENOMAD_DOWNLOAD.out.versions)
    }

    //
    // MODULE: Run geNomad end-to-end
    //
    GENOMAD_ENDTOEND(
        fasta_gz,
        ch_genomad_db
    )
    ch_versions                 = ch_versions.mix(GENOMAD_ENDTOEND.out.versions)

    emit:
    plasmid_tsv         = GENOMAD_ENDTOEND.out.plasmid_summary  // [ [ meta ], plasmid_summary.tsv ]
    virus_tsv           = GENOMAD_ENDTOEND.out.virus_summary    // [ [ meta ], virus_summary.tsv ]
    virus_fna_gz        = GENOMAD_ENDTOEND.out.virus_fasta      // [ [ meta ], virus.fna.gz ]
    genes_tsv           = GENOMAD_ENDTOEND.out.genes            // [ [ meta ], genes.tsv ]
    features_tsv        = GENOMAD_ENDTOEND.out.features         // [ [ meta ], features.tsv ]
    scores_tsv          = GENOMAD_ENDTOEND.out.scores           // [ [ meta ], aggregated_classification.tsv ]
    taxonomy_tsv        = GENOMAD_ENDTOEND.out.taxonomy         // [ [ meta ], taxonomy.tsv ]
    proteins_faa_gz     = GENOMAD_ENDTOEND.out.faa              // [ [ meta ], faa.gz ]
    versions            = ch_versions                           // channel: [ versions.yml ]
}
