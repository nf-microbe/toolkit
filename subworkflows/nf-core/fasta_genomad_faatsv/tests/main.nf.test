nextflow_workflow {

    name "Test Subworkflow FASTA_GENOMAD_FAATSV"
    script "../main.nf"
    workflow "FASTA_GENOMAD_FAATSV"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/fasta_genomad_faatsv"
    tag "subworkflows/utils_nfmicrobe_functions"
    tag "genomad"
    tag "genomad/download"
    tag "genomad/endtoend"

    test("sarscov2 - genome - fasta.gz") {
        when {
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test1' ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta.gz', checkIfExists: true)
                    ],
                    [
                        [ id:'test2' ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz', checkIfExists: true)
                    ]
                )
                input[1] = null
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
