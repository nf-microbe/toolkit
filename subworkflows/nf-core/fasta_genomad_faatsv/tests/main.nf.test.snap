{
    "sarscov2 - genome - fasta.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_plasmid_summary.tsv:md5,8c4ddaa8a90da779c11537be0fddb799"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_plasmid_summary.tsv:md5,951e9110c265af707b2c65500a7acacd"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_virus_summary.tsv:md5,923b36db276dfee18ac1f1b57cae617e"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_virus_summary.tsv:md5,e2f291c05ebe6cbeaca7a30cb845eb04"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_genes.tsv:md5,f84d8a9fce3c92c49e78672bea16561f"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_genes.tsv:md5,2cd0f7b98a3c880e7eea765aa821ff19"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_features.tsv:md5,868804cf1857d3c7e129086ec747c77c"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_features.tsv:md5,2de71bfdaa71a6a6ced6dc5160c6378c"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_aggregated_classification.tsv:md5,702aead49cca96c98e43c5f032f60b26"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_aggregated_classification.tsv:md5,52472567bef1698395d7afef49bd0350"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_taxonomy.tsv:md5,44c6cf5ff4c836f66648457ae8deb0d5"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_taxonomy.tsv:md5,22081fd09bee75003b5aac12f05b7a29"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_proteins.faa.gz:md5,6850bd46dcb921e5cfb23bfbb8fcc12b"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_proteins.faa.gz:md5,63d0544a5d8e1448d375e09329eec1b7"
                    ]
                ],
                "7": [
                    "versions.yml:md5,2a9074bb10605724a4986c6f52919839",
                    "versions.yml:md5,a9b7acb49920049fbcd519326fe63cf7",
                    "versions.yml:md5,a9b7acb49920049fbcd519326fe63cf7"
                ],
                "features_tsv": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_features.tsv:md5,868804cf1857d3c7e129086ec747c77c"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_features.tsv:md5,2de71bfdaa71a6a6ced6dc5160c6378c"
                    ]
                ],
                "genes_tsv": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_genes.tsv:md5,f84d8a9fce3c92c49e78672bea16561f"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_genes.tsv:md5,2cd0f7b98a3c880e7eea765aa821ff19"
                    ]
                ],
                "plasmid_tsv": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_plasmid_summary.tsv:md5,8c4ddaa8a90da779c11537be0fddb799"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_plasmid_summary.tsv:md5,951e9110c265af707b2c65500a7acacd"
                    ]
                ],
                "proteins_faa_gz": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_proteins.faa.gz:md5,6850bd46dcb921e5cfb23bfbb8fcc12b"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_proteins.faa.gz:md5,63d0544a5d8e1448d375e09329eec1b7"
                    ]
                ],
                "scores_tsv": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_aggregated_classification.tsv:md5,702aead49cca96c98e43c5f032f60b26"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_aggregated_classification.tsv:md5,52472567bef1698395d7afef49bd0350"
                    ]
                ],
                "taxonomy_tsv": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_taxonomy.tsv:md5,44c6cf5ff4c836f66648457ae8deb0d5"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_taxonomy.tsv:md5,22081fd09bee75003b5aac12f05b7a29"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2a9074bb10605724a4986c6f52919839",
                    "versions.yml:md5,a9b7acb49920049fbcd519326fe63cf7",
                    "versions.yml:md5,a9b7acb49920049fbcd519326fe63cf7"
                ],
                "virus_tsv": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_virus_summary.tsv:md5,923b36db276dfee18ac1f1b57cae617e"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_virus_summary.tsv:md5,e2f291c05ebe6cbeaca7a30cb845eb04"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-10T07:30:33.357738596"
    }
}